docs/project-versioning.md

```diff
@@ -18,7 +18,9 @@ More in depth:
 
 **PATCH** version when you make backwards-compatible bug fixes.
```

Comment:
IDK if this is in scope for this PR, but I think some examples might help here - here's some that I think might work:

**Major**
Examples:
* Major +/or high risk rewrites that are hard to test completely such as rewriting pathfinding
* Changing the process for upgrading/installing AzerothCore such as requiring Visual Studio 2019 instead of 2017 or adding a new dependency such as 'Boost'
* Removing GM commands or changing the behavior of existing GM commands
* Removing existing core hooks, or modifying existing core hooks in a way that will break existing modules unless maintainers make changes

**Minor**
Examples:
* Implementing a previously non-implemented boss fight/quest/event/etc
* Adding a patrol route for an NPC that used to incorrectly stand still
* Adding new core hooks, without changing existing hooks
* Adding new GM commands

**Patch**
Examples:
 * Moving an ore vein that was floating in midair
 * Fixing a bug that made a boss fight impossible
 * Fixing a critical defect that was causing a server crash during normal gameplay
 * Fixing a broken GM command

Reply:
As I said, we can't use major-minor-patch because, otherwise, we can potentially have dozens of major versions released every week.

Our strategy is (in order of most common actions): 
1) to not update the version when the changes are just fixes/chore changes
2) to update the `-dev.x` prerelease version on master when we have breaking changes or new features
3) to upgrade the major version when we decide to release a new stable version
4) if any new security patch or feature is imported within the released major version, those ones will increase the minor/patch version then, but we do it very occasionally.

AzerothCore is not yet considered a "completed" software at an enterprise level. The versioning described above will be used to provide a simple way to check if a module is compatible or not with AC and what to do to upgrade it.

In fact, an author module should always provide the information about what was the latest tested version (I'm creating another PR to facilitate this work to them by implementing an internal package-manager system similar to npm...it's almost done)